src/infrasys/time_series_models.py

```diff
@@ -89,6 +101,72 @@ def check_data(
 
         return data
 
+    @classmethod
+    def aggregate(cls, ts_data: list[Self]) -> Self:
```

Comment:
Perhaps `[Iterable]` instead of `list`?

Reply:
actually no mypy complains if I do iterable because I am accessing ts_data in multiple places which you can not do if you pass generator

Reply:
Ah, because you are accessing it by index. Technically, you could have made it `typing.Sequence`, which allows list or tuple, at least. But don't worry about it.